src/gfn/gym/hypergrid.py

```diff
@@ -131,17 +131,17 @@ def update_masks(self, states: DiscreteStates) -> None:
         )
         states.backward_masks = states.tensor != 0
 
-    def make_random_states_tensor(self, batch_shape: Tuple[int, ...]) -> torch.Tensor:
+    def make_random_states_tensor(
+        self, batch_shape: Tuple[int, ...], device: torch.device
+    ) -> torch.Tensor:
         """Creates a batch of random states.
 
         Args:
             batch_shape: Tuple indicating the shape of the batch.
 
         Returns the batch of random states as tensor of shape (*batch_shape, *state_shape).
         """
-        return torch.randint(
-            0, self.height, batch_shape + self.s0.shape, device=self.device
-        )
+        return torch.randint(0, self.height, batch_shape + self.s0.shape, device=device)
```

Comment:
Same comment about it coming from the  env default device if the user does not pass the device.

Reply:
ok thanks